src/lbaf/Execution/lbsAlgorithmBase.py

```diff
@@ -18,7 +18,12 @@ def __init__(self, work_model, parameters: dict, lgr: Logger, qoi_name: str=''):
             parameters: a dictionary of parameters
             qoi_name: optional additional QOI to track"""
 
-        # Assign logger to instance variable
+        # Assert that a logger instance was passed
+        if not isinstance(lgr, Logger):
+            lgr().error(
```

Comment:
Is the `lgr()` here still a problem?

Reply:
It shouldn't, because in this case we are calling the `lgr()` function (provided by the utilities) to give us a logger so we may throw the error somewhere.